model_zoo/lbann.cpp

```diff
@@ -102,6 +113,24 @@ int main(int argc, char* argv[])
       return EXIT_SUCCESS;
     }
 
+    auto const hang_rank = arg_parser.get<int>("hang on start");
+    if (hang_rank >= 0)
+    {
+      auto const my_rank = comm->get_rank_in_world();
+      if (hang_rank == my_rank) {
+#ifdef LBANN_HAS_UNISTD_H
+        char hostname[1024];
+        gethostname(hostname, 1024);
+        std::cout << "LBANN [hang]: (hostname: " << hostname << ", pid: " << getpid()
```

Comment:
Just as a note: The use of `std::cout` instead of `std::cerr` here is intentional. Due to the different buffering semantics between the two streams, I encountered situations in which this would it the console prior to a bunch of `std::cout` output. So I add it to that stream so it should be guaranteed to be the last thing to print prior to actually hanging, making it easily found by users. An alternative would be to flush `std::cout` and then push this to `std::cerr`.